one_fm/overrides/leave_application.py

```diff
@@ -90,12 +90,23 @@ def close_shifts(self):
         #delete the shifts if a leave application is approved
         try:
             if self.status == "Approved":
-                query =f"""DELETE from `tabShift Assignment` where employee = '{self.employee}' and start_date BETWEEN '{self.from_date}' and '{self.to_date}' and docstatus = 1 """
-                frappe.db.sql(query)
+                shift_assignment = frappe.db.sql(f"""SELECT name from `tabShift Assignment` where employee = '{self.employee}' and start_date BETWEEN '{self.from_date}' and '{self.to_date}' and docstatus = 1 """, as_dict=True)
+                for shift in shift_assignment:
+                    #unlink from attendance check
+                    self.unlink_attendance_check(shift.name)
+                    query =f"""DELETE from `tabShift Assignment` where name = '{shift.name}' and docstatus = 1 """
+                    frappe.db.sql(query)
                 frappe.msgprint(msg = f"Shift Assignments for {self.employee_name} between {self.from_date} and {self.to_date} have been deleted",alert=1)
         except:
             frappe.log_error(frappe.get_traceback(),"Error Closing Shifts")
 
+    def unlink_attendance_check(self, shift):
+        if frappe.db.exists("Attendance Check", {'shift_assignment':shift}):
+            att_check_doc = frappe.get_doc("Attendance Check", {'shift_assignment':shift})
+            att_check_doc.db_set('shift_assignment', "")
+            att_check_doc.db_set('has_shift_assignment',0)
```

Comment:
This can be write like
```
a_check_exists = frappe.db.exists("A Check",  {"shift_assignment": shift})
if a_check_exists:
  frappe.db.set_value("A Check", a_check_exists, "shift_assignment", "")
  frappe.db.set_value("A Check", a_check_exists, "has_shift_assignment", 0)
```

When we do `get_doc` for just getting the values of 1 or 2 fields, a lot of unnecessary data is being fetched. This becomes worse as the number of child table in a doctype increases

`frappe.db.set_value` sets a field's value in the database, does not call the ORM triggers but updates the modified timestamp (unless specified not to).

Reply:
Or You can set the values like,

```
frappe.db.set_value('A Check', a_check_exists, {
    'shift_assignment': '',
    'has_shift_assignment': 0
})
```